netbout-web/src/main/java/com/netbout/rest/TkInbox.java

```diff
@@ -73,33 +71,43 @@
 
     @Override
     public Response act(final Request req) throws IOException {
+        final String query = new RqHref.Smart(new RqHref.Base(req)).single(
+            "q", ""
+        );
         return new RsPage(
             "/xsl/inbox.xsl",
             this.base,
             req,
-            new XeAppend("bouts", this.bouts(req))
+            new XeAppend("bouts", this.bouts(req, query)),
+            new XeAppend("query", query),
+            new XeLink("search", new Href("/search"))
         );
     }
 
     /**
-     * All bouts in the inbox.
+     * Returns searched or paginated bouts in the inbox.
      * @param req Request
+     * @param query Search term
      * @return Bouts
      * @throws IOException If fails
      */
-    private Iterable<XeSource> bouts(final Request req) throws IOException {
-        long since = Inbox.NEVER;
-        final Iterator<String> param = new RqHref.Base(req).href()
-            .param("since").iterator();
-        if (param.hasNext()) {
-            since = Long.parseLong(param.next());
+    private Iterable<XeSource> bouts(final Request req, final String query)
+        throws IOException {
+        final Iterable<Bout> bouts;
+        final Inbox inbox = new RqAlias(this.base, req).alias().inbox();
+        if (StringUtils.isBlank(query)) {
+            long since = Inbox.NEVER;
+            final Iterator<String> param = new RqHref.Base(req).href()
+                .param("since").iterator();
+            if (param.hasNext()) {
+                since = Long.parseLong(param.next());
+            }
+            bouts = Iterables.limit(inbox.jump(since).iterate(), Inbox.PAGE);
+        } else {
+            bouts = inbox.search(query);
```

Comment:
@erimerturk yes, feel free  to create a ticket about the pageable result of the search
but `Iterables.limit` just limits the result here, 
I think it would better get partial result instead a possible crash of the browser, isn't it?